Gems/RecastNavigation/Code/Tests/NavigationMeshTest.cpp

```diff
@@ -647,4 +649,148 @@ namespace RecastNavigationTests
         RecastNavigation::RecastNavigationPhysXProviderCommon test(true);
         EXPECT_EQ(strcmp(test.TYPEINFO_Name(), "RecastNavigationPhysXProviderCommon"), 0);
     }
+
+    TEST_F(NavigationTest, Async)
+    {
+        Entity e;
+        PopulateEntity(e);
+        ActivateEntity(e);
+        SetupNavigationMesh();
+
+        ON_CALL(*m_mockPhysicsShape.get(), GetGeometry(_, _, _)).WillByDefault(Invoke([this]
+        (AZStd::vector<AZ::Vector3>& vertices, AZStd::vector<AZ::u32>& indices, const AZ::Aabb*)
+            {
+                AddTestGeometry(vertices, indices, true);
+            }));
+
+        const Wait wait(AZ::EntityId(1));
+        RecastNavigationMeshRequestBus::Event(e.GetId(), &RecastNavigationMeshRequests::UpdateNavigationMeshAsync);
+        wait.BlockUntilCalled(AZ::TimeMs{ 100 });
```

Comment:
question: Could any of these tests fail if running on hardware under load? Is this timeout long enough? Just want to make sure we avoid any flaky tests if we can help it 😅

Reply:
Hmm, so inside BlockUntilCalled I am actually running a loop with 5ms iteration. So if the whole process freezes we should still be good. But I can bump the value on that to a few seconds instead.

Reply:
Done.